modules/Bio/EnsEMBL/Compara/PipeConfig/Parts/OrthologQMAlignment.pm

```diff
@@ -0,0 +1,171 @@
+                '2->A' => [ 'prepare_orthologs' => INPUT_PLUS() ],
+                'A->1' => [ 'check_file_copy' ],
+            }
+        },
+
+        {   -logic_name => 'prepare_orthologs',
+            -module     => 'Bio::EnsEMBL::Compara::RunnableDB::OrthologQM::PrepareOrthologs',
+            -parameters => {
+                'hashed_mlss_id'            => '#expr(dir_revhash(#orth_mlss_id#))expr#',
+                'homology_flatfile'         => '#homology_dumps_dir#/#hashed_mlss_id#/#orth_mlss_id#.#member_type#.homologies.tsv',
+                'homology_mapping_flatfile' => '#homology_dumps_dir#/#hashed_mlss_id#/#orth_mlss_id#.#member_type#.homology_id_map.tsv',
+            },
+            -analysis_capacity  =>  50,  # use per-analysis limiter
+            -flow_into => {
+                # these analyses will write to the same file, so a semaphore is required to prevent clashes
+                '3->A' => [ 'reuse_wga_score' ],
+                'A->2' => { 'calculate_wga_coverage' => INPUT_PLUS() },
+            },
+            -rc_name  => '2Gb_job',
+        },
+
+        {   -logic_name => 'calculate_wga_coverage',
+            -module     => 'Bio::EnsEMBL::Compara::RunnableDB::OrthologQM::CalculateWGACoverage',
+            -hive_capacity => 30,
+            -batch_size => 10,
+            -flow_into  => {
+                3 => [ '?table_name=ortholog_quality' ],
+                2 => [ 'assign_wga_coverage_score' ],
+            },
+            -rc_name => '2Gb_job',
+        },
+
+        {   -logic_name => 'assign_wga_coverage_score',
+            -module     => 'Bio::EnsEMBL::Compara::RunnableDB::OrthologQM::AssignQualityScore',
+            -parameters => {
+                'hashed_mlss_id' => '#expr(dir_revhash(#orth_mlss_id#))expr#',
+                'output_file'    => '#wga_dumps_dir#/#hashed_mlss_id#/#orth_mlss_id#.#member_type#.wga.tsv',
+                'reuse_file'     => '#wga_dumps_dir#/#hashed_mlss_id#/#orth_mlss_id#.#member_type#.wga_reuse.tsv',
+            },
+            -hive_capacity => 400,
+        },
+
+        {   -logic_name => 'reuse_wga_score',
+            -module     => 'Bio::EnsEMBL::Compara::RunnableDB::OrthologQM::ReuseWGAScore',
+            -parameters => {
+                'hashed_mlss_id'            => '#expr(dir_revhash(#orth_mlss_id#))expr#',
+                'previous_wga_file'         => '#prev_wga_dumps_dir#/#hashed_mlss_id#/#orth_mlss_id#.#member_type#.wga.tsv',
+                'homology_mapping_flatfile' => '#homology_dumps_dir#/#hashed_mlss_id#/#orth_mlss_id#.#member_type#.homology_id_map.tsv',
+                'output_file'               => '#wga_dumps_dir#/#hashed_mlss_id#/#orth_mlss_id#.#member_type#.wga_reuse.tsv',
+            },
+            -hive_capacity => 400,
+        },
+
+        {   -logic_name  => 'check_file_copy',
+            -module      => 'Bio::EnsEMBL::Hive::RunnableDB::Dummy',
+            -parameters  => { 'orth_wga_complete' => 1, },
```

Comment:
I still don't understand how this param is making it into `pipeline_wide_parameters`. Again, this is a dummy analysis (so doesn't do anything) - the most passing this param here will do is allow `check_file_copy` and `copy_files_to_shared_loc` to see it (unless you're using some trick I don't know about).

Usually to update/write a pipeline_wide_param, you would need to set up a dataflow to write the param there (see https://github.com/Ensembl/ensembl-compara/blob/master/modules/Bio/EnsEMBL/Compara/PipeConfig/Parts/EpoLowCoverage.pm#L224) or use this piece of code directly in the runnable: https://github.com/Ensembl/ensembl-compara/blob/master/modules/Bio/EnsEMBL/Compara/RunnableDB/CreateReuseSpeciesSets.pm#L128-L131

Reply:
That's strange, I don't know. It worked though? If I am using any tricks, I also don't know about them.